src/Tools/CoreFx.Tools/GenerateSlnFile.cs

```diff
@@ -0,0 +1,497 @@
+                            }
+                            if (projFile.Contains("TestNativeService")) //Windows (Visual C++) 
+                            {
+                                slnGuid = "8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942";
+                            }
+                            if (projFile.Contains("WebServer.csproj")) //Web Application
+                            {
+                                slnGuid = "349C5851-65DF-11DA-9384-00065B846F21";
+                            }
+
+                            string projectName = Path.GetFileNameWithoutExtension(projFile);
+                            string relativePathFromCurrentDirectory = projFile.Replace(slnFile + Path.DirectorySeparatorChar, "");
+                            string projectDefn =
+                                $"Project(\"{{{slnGuid}}}\") = \"{projectName}\", \"{relativePathFromCurrentDirectory}\", \"{{{currentProject.projectId}}}\"\r\n";
+
+                            string projectDependency = string.Empty;
+                            if (currentProjectType.DependsOn != null)
+                            {
+                                foreach (var dependency in currentProjectType.DependsOn)
+                                {
+                                    if (mapOfProjectTypes[dependency].InUse)
+                                    {
+                                        List<SlnProject> depProjects = null;
+                                        mapOfProjectsToTypes.TryGetValue(dependency, out depProjects);
+                                        foreach (var depProject in depProjects)
+                                        {
+                                            string depProjectId = depProject.projectId;
+                                            projectDependency =
+                                                $"\tProjectSection(ProjectDependencies) = postProject\r\n\t\t{{{depProjectId}}} = {{{depProjectId}}}\r\n";
+                                        }
+                                    }
+                                }
+                                if (!string.IsNullOrEmpty(projectDependency))
+                                {
+                                    projectDependency += "\tEndProjectSection\r\n";
+                                }
+                            }
+
+                            if (headers.ContainsKey(projectType.Key))
+                            {
+                                headers[projectType.Key].Add(projectDefn + projectDependency + "EndProject\r\n");
+                            }
+                            else
+                            {
+                                headers.Add(projectType.Key, new List<string>() { projectDefn + projectDependency + "EndProject\r\n" });
+                            }
+                        }
+                    }
+                    if (transientMapOfProjectTypes.ContainsKey(projectType.Key))
```

Comment:
I don't really understand what this `transientMapOfProjectTypes` thing is for, a comment would help.